### ISSUE TITLE
Incorrect battery percentage value

### ISSUE THREAD

**Issue by Wyse-** [Roliga/g930-battery-percentage#1]
Greetings!
I managed to [port your software to Windows](https://github.com/Wyse-/g930-battery-percentage-win), and also tested the original one on Debian Stretch on a different machine. I get the same values on both of them, however they seem to be incorrect: the value sits at about 45 while my headset is turning itself off from low battery. Not sure if they can be useful, but here's a few screenshots of the values I'm getting:
- [On Debian](https://i.imgur.com/2GrZk7c.png)
- [On Windows, compared to the Logitech software](https://i.imgur.com/HV5VL2D.png)

The values don't seem to be totally off though, I believe I was getting ~67% from your software when the Logitech software was showing ~27%, which is a ~40% difference, similar to the one from 45 to a very low value.

**Comment by Roliga** [comment]
Thanks for the report! Neat to see some interest in this!

I can indeed reproduce this, my headset dies at 44-46% according to g930-battery-percentage. Since the battery in my G930s is near dead though I just assumed that was the problem.

I'll have a closer look at this when I'm at my workstation, but in the meantime would you mind checking what value g930-battery-percentage reports when your headset is fully charged, with and without the cable plugged in?

**Comment by Wyse-** [comment]
Sure thing! Right now I'm getting 79% while unplugged and 84-89% while plugged in. I believe it's not fully charged though, I'll report back when it is.

EDIT: I believe it's fully charged atm, I get 81% unplugged and 89-91% while plugged in.

**Comment by Roliga** [comment]
So the obvious quick and dirty solution to this is to just map the range of values we get to 0-100. I implemented this as a temporary solution here so the software is at least usable.

There's certainly some better way to do this though. [This](https://i.imgur.com/i2k7UwS.png)  is a chart with one line for each byte in the relevant hid report that we read during a 100% to 0% discharge cycle. Right now we're only using the 14th byte (*q* in the chart) which has a very obvious draining pattern, however the two bytes *p* and *o* before that also seem related, *p* for example gradually falls from 255 to 0 for each integer decrement of *q*. I'm sadly not very familiar with various ways numbers can be encoded but I'll see if I can figure something out.

[Here](https://gist.github.com/Roliga/bffe9a953e11e2e308a8fbe2ac88e387) is the data if you want to have a look at it yourself. Each row is one hid report, the first column is time in seconds followed by one column for each byte in the report.

**Comment by Wyse-** [comment]
Awesome, thanks for sharing your data! I implemented your fix on my Windows port and comparing it to the value read by the Logitech software ([here](https://i.imgur.com/xdAk54L.png)) it seems to be fairly accurate. 

Not even the Logitech software reads it as a 0-100 percentage though (or at least it doesn't store it in memory like that), so [I mapped it to 0-100](https://github.com/Wyse-/G930BatteryReader-CLIBar/blob/83fce0c5a7dac367f07d293d536b4008bc27f00b/main.c#L30) just like you did. 

Feel free to check out the last part of the FAQ [here](https://github.com/Wyse-/G930BatteryReader) if you're curious about how the Logitech software handles it.
